src/pgrepr/src/types.rs

```diff
@@ -568,7 +568,13 @@ impl Type {
         }
     }
 
-    /// Returns the name that PostgreSQL uses for this type.
+    /// Returns the item's name in a way that guarantees it's resolvable in the
+    /// catalog.
+    pub fn catalog_name(&self) -> &'static str {
+        self.inner().name()
+    }
```

Comment:
Would it be possible to change this to 
```rust
    pub fn catalog_name(&self) -> (&'static str, &'static str) {
        (self.inner().name(), self.inner().schema())
    }
```

That way when we go to look it up in the catalog, we can check the specific system schema and not have to look in all of them.

Does the `schema()` return accurate information? I see that it's part of an external library.

Reply:
That would work if we change the crates' dependency graphs and generated `ParamType` from builtin types (not a bad idea, just work that we could elect to do). In the current world `ParamType` isn't attached to all of the builtin types, so the pseudotype parameters don't have any notion of their schema.

SQL planning doesn't currently keep such a tight hold on e.g. system schemas, so I was trying to avoid muddying that relationship any more than I needed to.

This all being said, I think we're due for a refactor that brings builtin types into something that can be more easily shared between SQL and coord, for exactly this kind of inelegance.

Reply:
Well that sounds like a lot of work for probably not a ton of benefit as far as this specific PR's functionality is concerned. So unless you're really eager to do that, it's probably best left to a later issue/PR.